osquery/utils/system/linux/proc/proc.h

```diff
@@ -13,10 +13,31 @@
 
 #include <string>
 
+#include <osquery/utils/status/status.h>
+
 namespace osquery {
 namespace proc {
 
 std::string cmdline(pid_t pid);
 
+/*
+ * Use SYS_setns syscall to set namespace.
+ * Considers these situations:
+ *  - early in startup as shell if --setns flag set
+ *  - in shell when using `.setns` command
+ *  - running queries in container (in future ?)
+ *
+ * NOTE: setns() function not added until glibc 2.14
+ * Requires that process is single-threaded.
+ * To accommodate callers, his function does not log
+ * anything, but will provide INFO level in feedback
+ * and error message in status when failure.
+ *
+ * @param cpath The pid or path to switch to.  If cpath is
+ *              empty, will switch back to original namespace
+ *              of the host.  Path format: /proc/<pid>/ns/mnt
+ */
+Status setLinuxNamespace(const char* cpath, std::string& feedback);
```

Comment:
I think the `std::string& feedback` pattern is odd and I recommend trying something different. If we want to communicate something back to the caller I recommend an `enum` class with the expected outcomes.

Reply:
I recommend `const std::string& cpath` and avoiding `char*` when possible in osquery code.

Reply:
The `const char *` is used in the meta commands code (`devtools/shell.cpp`) ... I guess I can wrap it.